Terminal.Gui/Views/RuneCellEventArgs.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+
+namespace Terminal.Gui {
+	/// <summary>
+	/// Args for events that relate to a specific <see cref="RuneCell"/>.
+	/// </summary>
+	public class RuneCellEventArgs {
+		/// <summary>
+		/// The line that is currently drawn.
+		/// </summary>
+		public List<RuneCell> Line { get; set; }
+
+		/// <summary>
+		/// The col index position in the line.
+		/// </summary>
+		public int IdxCol { get; set; }
+
+		/// <summary>
+		/// The row index position in the text.
```

Comment:
The primary purpose of the event that uses this arg (e.g. `DrawNormalColor`) is to assign color by mutating the contents of `Line`.

It is imperative to know where within the `Text` the line is so that color can be assigned from a complex data structure (e.g. the results of parsing `Text`)

Where the rune is on the screen is of secondary importance.

I am not sure I like the mutateable events design though.  An alternative approach would be for the user subscribe to TextChanged and call methods like to change the cells color directly.  Maybe something like:


`ColorRunes( int startingAt, int length, ColorScheme withScheme)`

Reply:
Reading the comments here (https://github.com/gui-cs/Terminal.Gui/pull/2682#issuecomment-1579408525) you'll understand why I added the row parameter. It's not needed and can be retrieved by query using the `TopRow` property and the textview frame height. The Col property can be only read-only too. The main reason for using this event was only to process the visible current lines on the screen and not all the text, which may have a tons of lines.

Reply:
> It's not needed and can be retrieved by query using the TopRow property and the textview frame height. 

Won't that vary according to whether line wrap is on though? 

> Reading the comments here (https://github.com/gui-cs/Terminal.Gui/pull/2682#issuecomment-1579408525) you'll understand why I added the row parameter. It's not needed and can be retrieved by query using the TopRow property and the textview frame height.

You are right, I think I am over complicating things.  What matters most is that you can map the contents of the `Line` property to the original `Text`.  Which I think can be done without considering line number/rows and just having a single property.  Something like:

```csharp
/// <summary>
///  Index of the source text which <see cref="Line"> begins at.
///  Note that in some cases (e.g. word wrapping) this
///  may not correspond to start of a new line in the source.
/// </summary>
public int StartIndex {get;}
```

Reply:
> > It's not needed and can be retrieved by query using the TopRow property and the textview frame height.
> 
> Won't that vary according to whether line wrap is on though?
> 
Yes it vary. The `GetAllLines` method will return all lines according to the `WordWrap` property. I'll try to map it to the unwrapped text.

> > Reading the comments here ([#2682 (comment)](https://github.com/gui-cs/Terminal.Gui/pull/2682#issuecomment-1579408525)) you'll understand why I added the row parameter. It's not needed and can be retrieved by query using the TopRow property and the textview frame height.
> 
> You are right, I think I am over complicating things. What matters most is that you can map the contents of the `Line` property to the original `Text`. Which I think can be done without considering line number/rows and just having a single property. Something like:
> 
> ```cs
> /// <summary>
> ///  Index of the source text which <see cref="Line"/> begins at.
> ///  Note that in some cases (e.g. word wrapping) this
> ///  may not correspond to start of a new line in the source.
> /// </summary>
> public int StartIndex {get;}
> ```

You aren't complicating at all. We only are discussing the better way to do this. I'll consider to map the row always with the unwrapped text and it'll make more sense. The Line and Col will be the same because these two are needed into this event as is.

Reply:
I hope you like the solution I made to maintain a relationship between the Line and Col (wrapped or unwrapped) and with the `UnwrappedPosition` which is always the unwrapped line and col.
Fixes with the https://github.com/gui-cs/Terminal.Gui/pull/2682/commits/305c616c8469d72142dc6384771891118a3b1e56.